Source/JavaScriptCore/API/JSValueRef.h

```diff
@@ -292,6 +320,140 @@ JS_EXPORT JSValueRef JSValueMakeString(JSContextRef ctx, JSStringRef string);
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first and then compare with the integer.
+*/
+JS_EXPORT JSRelationCondition JSValueCompareUInt64(JSContextRef ctx, JSValueRef left, uint64_t right, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Compares a JSValue with a signed 64-bit integer.
+    @param ctx        The execution context to use.
+    @param left       The JSValue as the left operand.
+    @param right      The int64_t as the right operand.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first and then compare with the integer.
+*/
+JS_EXPORT JSRelationCondition JSValueCompareInt64(JSContextRef ctx, JSValueRef left, int64_t right, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Compares a JSValue with a double.
+    @param ctx        The execution context to use.
+    @param left       The JSValue as the left operand.
+    @param right      The double as the right operand.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first and then compare with the double.
+*/
+JS_EXPORT JSRelationCondition JSValueCompareDouble(JSContextRef ctx, JSValueRef left, double right, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Compares two JSValues.
+    @param ctx        The execution context to use.
+    @param left       The JSValue as the left operand.
+    @param right      The JSValue as the right operand.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       The result is computed by `IsLooselyEqual` and `IsLessThan` in JavaScript. If either the left or right JSValue has a NaN result of `ToPrimitive(value, NUMBER)`, a kJSRelationConditionUndefined is returned.
```

Comment:
The result is computed by comparing the results of `left == right`, `left < right`, and `left > right` in JavaScript. If either `left` or `right` is (or would coerce to) `NaN` then the result is `kJSRelationConditionUndefined`.